resolve/resolve.go

```diff
@@ -117,13 +122,13 @@ func File(file *syntax.File, isPredeclaredGlobal, isUniversal func(name string)
 // Expr resolves the specified expression.
 // It returns the local variables bound within the expression.
 //
-// The isPredeclaredGlobal and isUniversal predicates report whether a
-// name is a pre-declared global identifier (in the current module) or a
-// universal identifier (in every module).
-// Typical values are globals.Has and Universe.Has, respectively, where
-// globals is the current module's StringDict.
-func Expr(expr syntax.Expr, isPredeclaredGlobal, isUniversal func(name string) bool) ([]*syntax.Ident, error) {
-	r := newResolver(isPredeclaredGlobal, isUniversal)
+// The isPredeclared and isUniversal predicates report whether a name is
+// a pre-declared identifier (visible in the current module) or a
+// universal identifier (visible in every module).
+// Typical values are predeclared.Has and Universe.Has, respectively, where
+// predeclared is the module's StringDict of predeclared names.
+func Expr(expr syntax.Expr, isPredeclared, isUniversal func(name string) bool) ([]*syntax.Ident, error) {
```

Comment:
Does it still make sense to allow clients to modify `Universe`? The comments in library.go make it sound very similar to what *predeclared* is now.

I guess you still need to accept predicates for both here to avoid a circular dependency.

Reply:
As you point out, the intention here is not to allow clients to modify Universe, but to break a dependency.  (The new design obviates any need for clients to modify Universe, although it may yet be convenient to do so.)